### PR TITLE
feat: static application details page

### PR DESCRIPTION
**Pull request by admy7**
Static page for application details page

The sidebar for details is a bit empty. I will probably include more details about Datasets, Applicant and Events once it is integrated with Rems.

Screenshot 2024-03-27 at 11 54 51

Updated view: 

Screenshot 2024-03-27 at 15 22 00

### CODE REVIEW

File: src/components/applicationDetailsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+function ApplicationDetailsContainer() {
+  return (
+    <div className="flex flex-col gap-y-8 ">
```
Comment (brunopacheco1) [inline src/components/applicationDetailsContainer.tsx:7]: important: the sidebar grey background must not expand to the column size, it must fit to the size of the content of itself.

File: src/components/fieldAttachmentContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}: FieldAttachmentContainerProps) {
+  return (
+    <div className="mt-10 rounded border-2 border-white-smoke p-6">
+      <div className="flex justify-between">
+        <div>
+          <h3 className="text-2xl text-primary">{`${fieldName} Attachment`}</h3>
+        </div>
+        <Button
+          type="info"
+          text="Upload files"
```
Comment (brunopacheco1) [inline src/components/fieldAttachmentContainer.tsx:24]: important: the endpoint supports single file upload only, thus it has to be in singular.

File: src/app/applications/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}: ApplicationDetailsPageProps) {
+  const { id } = params;
+  return (
+    <div className="mt-20 grid grid-cols-12 gap-x-20">
+      <div className="col-span-6 col-start-3">
+        <div className="px-3">
+          <div className="flex justify-between">
+            <h1 className="text-3xl text-primary">Application Title</h1>
+            <div className="flex gap-x-3">
+              <Button type="warning" text="Save changes" className="text-xs" />
```
Comment (brunopacheco1) [inline src/app/applications/[id]/page.tsx:24]: important: I really liked icons on the left side of action buttons - I think all of them have. Can you please add to these two as well?

**Review by brunopacheco1 — COMMENTED** [review]
important: the page is not responsive yet.

File: src/app/applications/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+export default function ApplicationDetailsPage({
+  params,
+}: ApplicationDetailsPageProps) {
+  const { id } = params;
+  return (
+    <div className="mt-20 grid grid-cols-12 gap-x-20">
+      <div className="col-span-6 col-start-3">
+        <div className="px-3">
+          <div className="flex justify-between">
+            <h1 className="text-3xl text-primary">Application Title</h1>
```
Comment (inderps) [inline src/app/applications/[id]/page.tsx]: can you use PageHeading component here for the heading?

File: src/components/applicationDetailsContainer.tsx (hunk 1 above)
Comment (inderps) [inline src/components/applicationDetailsContainer.tsx:7]: Actually for consistency, you can use the same [sidebar](https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/blob/main/src/app/datasets/%5Bid%5D/Sidebar.tsx) component as we have in dataset page. We could think of extracting it out as generic component where content can be passed as child

**Review by brunopacheco1 — DISMISSED** [review]

File: src/app/applications/[id]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  params,
+}: ApplicationDetailsPageProps) {
+  const { id } = params;
+  const sidebarItems = createApplicationSidebarItems();
+  return (
+    <div className="mt-20 grid grid-cols-12 gap-x-20">
+      <div className="col-span-6 col-start-3">
+        <div className="px-3">
+          <div className="flex justify-between">
+            <h1 className="text-3xl text-primary">Application {id}</h1>
```
Comment (inderps) [inline src/app/applications/[id]/page.tsx:25]: this could have been PageHeading component

**Review by inderps — DISMISSED** [review]

**Review by inderps — APPROVED** [review]

**Pull request merged**